File: examples/StarterRepo/model/spaces.py
from .types import DummyType1, DummyType2
from typing import TypedDict

TerminatingSpace = TypedDict('Terminating Space', {})
EmptySpace = TypedDict('Empty Space', {})
DummySpace1 = TypedDict('Dummy Space 1', {'a': DummyType1})
DummySpace2 = TypedDict('Dummy Space 2', {'a': DummyType1, 'b': DummyType1, 'c': DummyType2})

File: examples/StarterRepo/julia_model/spaces.jl
include("../../../../cadCAD.jl/src/spaces.jl")
include("types.jl")
using .Spaces: generate_space_type

generate_space_type((a=DummyType1,), "Dummy Space 1")
generate_space_type((a=DummyType1, b=DummyType1, c=DummyType2,), "Dummy Space 2")
